dash/scripts/docker-testing/mpich/dockerfile

```diff
@@ -18,10 +18,9 @@ WORKDIR       /tmp
 ENV           CC=gcc
 ENV           CXX=g++
 
-# Install PAPI from source
-# Compiler error when compiling with -pedantic for papi > 5.4.1
-# ADD         http://icl.utk.edu/projects/papi/downloads/papi-5.5.1.tar.gz papi.tgz
-ADD           http://icl.cs.utk.edu/projects/papi/downloads/papi-5.4.1.tar.gz papi.tgz
+# Compiler error when compiling with -pedantic for papi >= 5.5
+ ADD         http://icl.utk.edu/projects/papi/downloads/papi-5.5.1.tar.gz papi.tgz
+#ADD           http://icl.cs.utk.edu/projects/papi/downloads/papi-5.4.3.tar.gz papi.tgz
```

Comment:
Are we not building with -pedantic anymore?

Reply:
As the flag is not listed in the `compile_commands.json`, someone must have removed it. However we could compile with `-Wpedantic`, so only warnings are generated. Probably we should discuss this, but this exceeds this PR. Anyway, I'm not a fan of using pedantic as this often clashes with third-party libraries. And at least for PAPI we are in a dead-end.

The OpenMPI 3.0.0 tests are currently not enabled in CI, as this would require a third CI container (currently mpich, openmpi2). As we only have 4 containers, this would allow us only to execute one CI job at a time (instead of 2).

Our tests seem to work also with OpenMPI3, except some errors in the HDF5 API.